scripts/commit-emoji-helper.sh

```
@@ -0,0 +1,267 @@
+# Function to validate commit message
+validate_message() {
+    local message="$1"
+    
+    if [[ -z "$message" ]]; then
+        echo -e "${RED}❌ Error: Please provide a commit message${NC}"
+        return 1
+    fi
+    
+    # Check format: emoji type: description
+    if echo "$message" | grep -qE '^[^[:space:]]+ [a-z]+: .+'; then
+        local emoji=$(echo "$message" | sed -E 's/^([^[:space:]]+) [a-z]+: .*/\1/')
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)